aidants_connect_web/models.py

```diff
@@ -240,10 +241,15 @@ class Mandat(models.Model):
 
 class AutorisationQuerySet(models.QuerySet):
     def active(self):
-        return self.exclude(expiration_date__lt=timezone.now())
+        return self.exclude(expiration_date__lt=timezone.now()).filter(
+            revocation_date__isnull=True
+        )
 
     def expired(self):
-        return self.exclude(expiration_date__gt=timezone.now())
+        return self.filter(
```

Comment:
Est ce que ce ne serait pas plus simple de vérifier que le mandat n'est pas expiré ?
```suggestion
        return self.mandat.is_expired()
```

Reply:
ici c'est un queryset donc on ne manipule pas une instance donnée. et le mandat n'a pas l'info sur le fait que l'attestation ait été révoquée ou pas. j'ai aussi ajouté un `is_expired` sur le mandat (cf message plus bas)

Reply:
Je veux bien qu'on en parle de vive voix, je crois que j'ai du mal à comprendre 🤔

Reply:
je voulais dire qu'ici ces méthodes reviennent à augmenter les queries au niveau du modèle `Autorisation`,  le `self` ici représente le modèle (`Autorisation.objects.expired()`), pas une instance qui aurait elle un mandat.
Dans les commits suivants tu peux voir la version à jour de cette méthode:
```
def expired(self):
        return self.filter(
            Q(mandat__expiration_date__lt=timezone.now()) | (
                Q(mandat__expiration_date__gt=timezone.now()) & Q(revocation_date__isnull=False)
            )
        )
```